### PR TITLE
WIP: Format failed external asserts nicer

### PR DESCRIPTION
**Pull request by tryzniak**
Hello. This for #1277 . I'd love to show you what I've been up to, here are the new output
image
[testfile](https://gist.github.com/tryzniak/56ab30808be43db4891aa67f9275160f)

I don't really like the way `supertest` errors are shown, I have no idea how we could detect what line of the test its expect failed.
Other errors have inconsistent formatting, which is not very nice.
I'm thinking of capturing and detecting a type of an error earlier.
Any input would be useful for me! Thanks

### CODE REVIEW

File: lib/test.js [hunk 1]
```diff
@@ -8,6 +8,14 @@ const assert = require('./assert');
 const nowAndTimers = require('./now-and-timers');
 const concordanceOptions = require('./concordance-options').default;
 
+function isExternalAssertError(error) {
+	if (error.matcherResult) {
+		return error.matcherResult.actual && error.matcherResult.expected;
+	}
+
+	return error.actual && error.expected;
```
Comment (novemberborn) [inline lib/test.js]: Could you add comments clarifying which errors from which frameworks you're looking for here?

Comment (tryzniak) [reply lib/test.js]: Absolutely!

**Comment by tryzniak** [comment]
> Oh that's an interesting approach. You're looking at the thrown error to see if it may have come from an assertion library? And then the hope is that its `message` is useful?

Yeah! Judging by a quick look, they usually have an already formatted `message`.

**Comment by novemberborn** [comment]
> > Oh that's an interesting approach. You're looking at the thrown error to see if it may have come from an assertion library? And then the hope is that its `message` is useful?
> 
> Yeah! Judging by a quick look, they usually have an already formatted `message`.

Sounds great!

File: lib/test.js (hunk 1 above)
Comment (novemberborn) [reply lib/test.js]: Nah those can be added in a future PR if necessary.

**Comment by novemberborn** [comment]
Closing due to inactivity. We'd be happy to pick this up again when you have the time to get back to this.

**Pull request closed without merging**